source/server/http/admin.h

```diff
@@ -31,6 +31,30 @@
 namespace Envoy {
 namespace Server {
 
+/**
+ * Implementation of Server::AdminStream.
+ */
+class AdminStreamImpl : public AdminStream {
+public:
+  AdminStreamImpl(Http::StreamDecoderFilterCallbacks& callbacks, Http::HeaderMap& request_headers,
```

Comment:
It's worth commenting here that the ctor args are held by reference and must be held in a data structure that outlasts the stream.  Another thought I had was to have the AdminFilter *own* the AdminStreamImpl as a std::unique_ptr, which and this would be the only location for these objects.  That might be easier to reason about in terms of correctness.  It would  populate the AdminStreamImpl in the same place it does now.

Reply:
ok, will look into it, I just need to check with the Hystrix implementation that it doesn't break anything

Reply:
I'm not super opinionated about who needs to own the admin stream since either way it will be possible to be left with a dangling reference without explicitly destroying something IIUC (in once case it will be the stream, in the other it will be the callbacks).  However, as @jmarantz suggests, in either case, please document how one would know that the reference is invalidated (using `onDestryCallabacks()`).